src/VisualStudio/Core/Def/Implementation/TableDataSource/Suppression/VisualStudioSuppressionFixService.cs

```diff
@@ -382,8 +382,8 @@ private bool ApplySuppressionFix(IEnumerable<DiagnosticData> diagnosticsToFix, F
             // Kick off diagnostic re-analysis for affected projects so that diagnostics gets refreshed.
             Task.Run(() =>
             {
-                var uniqueProjectIds = diagnosticsToFix.Where(d => d.ProjectId != null).Select(d => d.ProjectId).Distinct();
-                _diagnosticService.Reanalyze(_workspace, uniqueProjectIds);
+                var reanalyzeDocuments = diagnosticsToFix.Where(d => d.DocumentId != null).Select(d => d.DocumentId).Distinct();
+                _diagnosticService.Reanalyze(_workspace, documentIds: reanalyzeDocuments, highPriority: true);
```

Comment:
Btw, I think it might help also to call RemoveDocument explicitly for these documents up front here, and then invoke Reanalyze - this way the fixed diagnostics should disappear immediately from error list, and they can appear later (soon enough) on high priority re-analysis.